WIP Eckhart recovery

bieleluk opened the pull request:
This PR implements the following components for Eckhart layout:
- `show_checklist` FirmwareUI trait function
- `continue_recovery flow`
- all recovery warning and success screens

obrusvit reviewed (changes requested):
A few points:
- put 'x' on left buttons where we cannot go back (number input for num of shares and threshold)
- checklist should either have a menu with cancel (if there will be some more info) or double button action bar (if there is nothing other than cancel in the menu?) I raised a point in Figma
- number input menu button cancels the flow
- TR strings: put the comma back and let's put it in the same font for now
- inconsistent hint at the 1st word - it disappears if you go back, should it even be there?

bieleluk commented:
@obrusvit 

- [x] put 'x' on left buttons where we cannot go back (number input for num of shares and threshold)
- [x]  checklist should either have a menu with cancel (if there will be some more info) or double button action bar (if there is nothing other than cancel in the menu?)
- [x] number input menu button cancels the flow
https://github.com/trezor/trezor-firmware/pull/4804/commits/1ca37220fb1566db8324176d78abeecfa75a7b2d
- [x] TR strings: put the comma back and let's put it in the same font for now
https://github.com/trezor/trezor-firmware/pull/4804/commits/cd4e591f974c4be2f2131f9543c4db5420576c24
- [x] inconsistent hint at the 1st word - it disappears if you go back, should it even be there?
https://github.com/trezor/trezor-firmware/pull/4804/commits/389077f95e473acedcb7a6256f8950dda7423d6b

Other minor improvements:
- correct placement of word number tiles:
https://github.com/trezor/trezor-firmware/pull/4804/commits/177b0e26c854fe3fbc8ebd844701c34b4034e375
- reverted timeout for the success screen and made it common for all layouts (the warnings must be layout-specific because they are too different for each layout)
https://github.com/trezor/trezor-firmware/pull/4804/commits/a107d1a9a3cb86ac11aa7699d8b685efdd38ab00
- fixed rendering of multi-digit numbers
 https://github.com/trezor/trezor-firmware/pull/4804/commits/9cf94551ec6a7c96564a15f00a6d3a776b129051
- updated warning titles according to the latest Figma update
https://github.com/trezor/trezor-firmware/pull/4804/commits/002be2eb572585dab8a62947040da8bbfb04b560
